### PR TITLE
79 packaging and coordinate renaming

### LINKED ISSUE #79: Update publishing to new org/module
## Description

There is no advantage to publishing the built connector jar to Maven central because it is typically not used as a dependency for other projects. Instead it is deployed to a plugin path in Kafka.

As such since we are moving the coordinates anyway it makes more sense to just publish the jar alongside the release on GitHub.

- [ ] Update publication names in gradle files
- [ ] Update Jenkinsfile publication process
- [ ] Update README instructions locations
- [ ] Fix packaging of jar
- [ ] rename packages to suit new coordinates

### PR DESCRIPTION
**Pull request by ricellis**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Rename packages and coordinates to align with GitHub org move.

Fixes #79 

## Approach

Rename `com.ibm.cloudant.kafka` to `com.ibm.cloud.cloudant.kafka` to align with new coordiantes.
Package jar as `cloudant-kafka-connector-<semver>.jar` and don't package kafka dependencies or include Kafka version in naming.
Remove unnecessary maven publishing requirements (javadoc & source jars & signing).
Update Jenkinsfile to declarative syntax and publish to GitHub release instead of maven.
Update documentation to match.

## Schema & API Changes

- Connector config needs to refer to new package names. This is alongside other breaking changes in proposed `0.200.0` version.

## Security and Privacy

- "No change"

## Testing

- N/A build or packaging only changes
- Manually tested tag release process - see [draft release](https://github.com/IBM/cloudant-kafka-connector/releases/tag/untagged-5340d23bd52aa37d5c83)

## Monitoring and Logging

- Loggers change name because of packaging changes.

### CODE REVIEW

**Review by tomblench — APPROVED** [review]
Connector seems to start up ok after tweaking properties file, +1

File: CHANGES.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # UNRELEASED
+- [BREAKING CHANGE] Publish releases to https://github.com/IBM/cloudant-kafka-connector/releases.
+- [BREAKING CHANGE] Rename package from `com.ibm.cloudant.kafka` to `com.ibm.cloud.cloudant.kafka`.
```
Comment (tomblench) [inline CHANGES.md:3]: I wonder if we should specifically call this out and mention that users need to tweak their `connector.class` in any existing properties files.

Comment (ricellis) [reply CHANGES.md:3]: I can add more description here.
Given the list of breaking changes between what we have now and what is slated for `0.200` - I figure we'll need some kind of migration guide anyway.

Comment (tomblench) [reply CHANGES.md:3]: TBH I think you're right - we will have a tonne of stuff to write when the next release happens so I don't think it's a blocker right now.

Comment (ricellis) [reply CHANGES.md]: Added in 2a90c8f4858648f90043ad8a8b0d9f32e4f59aa8

**Pull request merged**